internal/git/cli.go

```diff
@@ -424,6 +456,19 @@ func (g *GitCLI) RemoteAdd(name string, url string) error {
 	return g.run("remote", "add", name, url)
 }
 
+func (g *GitCLI) Merge(repositoryURL, baseSHA, headSHA, committerDate, workingDir string) (string, error) {
+	if g.mergeBinary == "" {
+		return "", fmt.Errorf("merge binary not found")
+	}
+
+	stdout, err := g.runMerge("merge", "--clone-url", repositoryURL,
```

Comment:
you pass the working directory but do not use it. Is the contract that the merge be done in the working directory? How will that work if the checkout action is being used against a non-root path?

The merge command to fill in for SCM providers that do not expose a consistent merge commit should either be doing its work in a completely isolated location and just returning the fetch url to allow the merge commit to be pulled from... or it should be being passed the git directory to work in... the latter runs the risk of git version mismatch, whereas treating the merge with its own git toolchain as a "remote" is the safer option.

In any case I suspect the implementation here will not work for cases where the action's default `path` has been overridden. and I have a suspicion that the action will also fail if the pull request workflow is trying to checkout a repository from another repo as it is unclear to me if the pull event path is sufficiently guarded.

The pull request path should only be triggered if the `repository` matches the event *and* the `ref` matches the event... in that case we should use the merge binary to get access to a remote with the merge commit (if not already created by a previous checkout step... think a user who decided to check out the source more than once in the same job) and pull that into the checkout's repo before continuing to check out the ref

Reply:
@stephenc - Addressed all your concerns above, and tested with a Bitbucket workflow. Following scenarios were verified:

1. Checkout a different repo within a path and verify contents.
2. Checkout the source repo, different branch on a different path and verify contents.
3. Checkout the source repo at the merge commit in the root path and verify contents.

Please re-review, thank you!

Reply:
I think this is defining too large a API contract surface for my tastes. I think we can have a simpler contract. Let's discuss when we're in compatible TZs